backend/pkg/httpserver/list_aggregated_feature_support.go

```diff
@@ -35,9 +57,23 @@ func (s *Server) ListAggregatedFeatureSupport(
 		return cachedResponse, nil
 	}
 
+	var targetMobileBrowser *string
+	if request.Params.IncludeBaselineMobileBrowsers != nil {
+		var err error
+		targetMobileBrowser, err = getDesktopsMobileProduct(request.Browser)
+		if err != nil {
+			return backend.ListAggregatedFeatureSupport400JSONResponse{
+				Code:    400,
+				Message: err.Error(),
+			}, err
+		}
+	} else {
+	}
```

Comment:
You should revert getDesktopsMobileProduct back to returning `backend.BrowserPathParam, error`. It loses the type and prevents other consumers from keeping the exact type.

Once you do that, you could do :

```suggestion
	var targetMobileBrowser *string
	if request.Params.IncludeBaselineMobileBrowsers != nil {
		matchingMobileBrowser, err := getDesktopsMobileProduct(request.Browser)
		if err != nil {
			return backend.ListAggregatedFeatureSupport400JSONResponse{
				Code:    400,
				Message: err.Error(),
			}, err
		}
		targetMobileBrowser = (*string)(&matchingMobileBrowser)
	}
```